src/ape/api/providers.py

```diff
@@ -84,25 +84,16 @@ def total_transfer_value(self) -> int:
     @abstractmethod
     def serialize_transaction(self) -> bytes:
         """
-        Take this object and produce a hash to sign to submit a transaction
+        Serialize the transaction
         """
 
-    def as_dict(self, exclude: Optional[Set[str]] = None) -> dict:
-        """
-        Create a ``dict`` representation of the transaction.
-
-        Returns:
-            dict
-        """
-        return self.dict(by_alias=True, exclude_none=True, exclude=exclude)
-
     def __repr__(self) -> str:
-        data = self.as_dict()  # NOTE: `as_dict` could be overridden
+        data = self.dict()
         params = ", ".join(f"{k}={v}" for k, v in data.items())
         return f"<{self.__class__.__name__} {params}>"
 
     def __str__(self) -> str:
-        data = self.as_dict()  # NOTE: `as_dict` could be overridden
+        data = self.dict()
```

Comment:
Maybe we should call this `txn_dict` to not confuse with a Transaction's data property

Reply:
Had a discussion around this and since we are no longer wrapping it as custom, I think it's okay to leave it as the pydantic model's dict method.